### ISSUE TITLE
Introduce Type Alias for Patterns and more

### ISSUE THREAD

**Issue by noklam** [kedro-org/kedro#2776]
# Background
As a follow up action, we agreed to introduce type like `Patterns = Dict[str, Dict[str, Any]]` to improve the readability of the code a lot. 

An example can be found [here](https://github.com/kedro-org/kedro/pull/2770/files#diff-83c76b6d924034cc585ba09c9fcb10161887b633b74e8b09a91e4803200f0ed9R33)

We can also do this beyond just `Patterns` but to more complicated types within the code base.

> 
>   As discussed, let's separate the Type Alias proposed in #2770.
>  I will close #2770 and approve this PR now. :)
> 
> _Originally posted by @noklam in https://github.com/kedro-org/kedro/issues/2635#issuecomment-1623710496_
>

**Comment by noklam** [comment]
This could be nicely rewritten to:
```
matches = (parse(pattern, data_set_name) for pattern in data_set_patterns.keys())
return next(filter(None, matches), None)
```

_Originally posted by @idanov in https://github.com/kedro-org/kedro/pull/2635#discussion_r1255521107_
            

There are some minor refactoring work linked here, we can address this with this change.

**Comment by noklam** [comment]
Closed with #2779
